### PR TITLE
Assorted error handling fixes

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds error handling in a few places to avoid triggering PHP Fatals, and updates the tests so they don't rely on network.

- Improves handling of a variable. `Newspack_Newsletters_Constant_Contact_SDK::get_contact` can return `false` or `WP_Error` in case of failure, and the `upsert_contact` method was not handling the latter case. 
- Improves error handling of Mailchimp usage reports, if the API key is not provided
- Updates the `test_render_embed_blocks` test not to rely on the network
- Improves error handling in ActiveCampaign usage reports, in case of a network error
- Fixes the failed CI builds (#1642)

Submitting as a hotfix because the lack of error handling triggers PHP Fatals, which pollute our logs. 

### How to test the changes in this Pull Request:

- Smoke test, this should be self-explanatory. 
- Observe all tests are passing

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-newspack-newsletters-renderer.php [hunk 1]
```diff
@@ -1012,8 +1012,8 @@ function ( $block ) {
 			 * Embed block.
 			 */
 			case 'core/embed':
-				$oembed = _wp_oembed_get_object();
-				$data   = $oembed->get_data( $attrs['url'] );
+				$oembed = apply_filters( 'newspack_newsletters_get_oembed_object', _wp_oembed_get_object() );
```
Comment (leogermani) [inline includes/class-newspack-newsletters-renderer.php:1020]:
```suggestion
				/**
				 * Filters the retrieval of the WP oEmbed object. Used for testing purposes
				 *
				 * @param WP_oEmbed $oembed WP_oEmbed object.
				 */
				$oembed = apply_filters( 'newspack_newsletters_get_oembed_object', _wp_oembed_get_object() );
```

**Review by leogermani — APPROVED** [review]
Nice.

I also see these errors when running tests. Not introduced by this PR, just sharing because it seems related

```
wp_die() called
Message: Error retrieving Mailchimp campaign.
Title: WordPress &rsaquo; Error
Args:
	response: 500
	code: wp_die
	exit: 1
	back_link: 
	link_url: 
	link_text: 
	text_direction: ltr
	charset: UTF-8
	additional_errors: array (
)
	newspack_newsletters_mailchimp_error: 
```

**Comment by dkoo** [comment]
@adekbadek @leogermani please look at https://github.com/Automattic/newspack-newsletters/pull/1642 to see if this fixes the timeouts we've been seeing on the `release` and `build-distributable` jobs.

**Comment by adekbadek** [comment]
@leogermani – the tests issue should also be resolved by this PR. In the last few failed CI test-php jobs, the offending test was `test_render_embed_blocks`, which is fixed here.

**Pull request merged**